common/src/services/user.service.ts

```diff
@@ -229,4 +249,29 @@ export class UserService implements UserServiceAbstraction {
     async clearProviders(userId: string): Promise<any> {
         await this.storageService.remove(Keys.providersPrefix + userId);
     }
+
+    async buildVerificationRequest<T extends PasswordVerificationRequest>
```

Comment:
I think we should move this to a different service.

Reply:
Added `userVerificationService` as discussed - thanks for the suggestion.